openpype/hosts/max/api/lib.py

```diff
@@ -323,17 +324,92 @@ def is_headless():
 
 @contextlib.contextmanager
 def viewport_camera(camera):
+    """Function to set viewport camera during context
+    ***For 3dsMax 2024+
+    Args:
+        camera (str): viewport camera for review render
+    """
     original = rt.viewport.getCamera()
+    has_autoplay = rt.preferences.playPreviewWhenDone
+    nitrousGraphicMgr = rt.NitrousGraphicsManager
+    viewport_setting = nitrousGraphicMgr.GetActiveViewportSetting()
+    orig_preset = viewport_setting.ViewportPreset
     if not original:
         # if there is no original camera
         # use the current camera as original
         original = rt.getNodeByName(camera)
     review_camera = rt.getNodeByName(camera)
     try:
         rt.viewport.setCamera(review_camera)
+        rt.preferences.playPreviewWhenDone = False
         yield
     finally:
         rt.viewport.setCamera(original)
+        rt.preferences.playPreviewWhenDone = has_autoplay
+        viewport_setting.ViewportPreset = orig_preset
+        rt.completeRedraw()
```

Comment:
I'm confused why `viewport_camera` context manager would ALSO tweak viewport setting presets and `playPreviewWhenDone` preferences. Those should not be inside this context manager?

Reply:
but if I call out rt.preferences.playPreviewWhenDone = False, this is actually changing the scene too. And actually for Max 2024, we can use autoPlay function to tackle this but for Max2023, it needs to be in contextmanager as it needs to be False(by default True)during context.

Reply:
I'm not sure what auto play does - but I assume it has nothing to do with 'setting the camera' right? Or does it auto play the scene when **setting** the camera?

Note that wherever you want to disable play preview when done and set the camera you can of course nest context managers:
```python
with play_preview_when_done(False):
    with viewport_camera(camera):
        # Ensure view is updated to the new camera
        rt.completeRedraw()
        publish()
```